### PR TITLE
Fix CMD Config and modify --historical-slasher-node 

### LINKED ISSUE #6702: beaconchaindata grows large very fast
# 🐞 Bug Report

### Description

Beaconchaindata grows large very fast and occupies more than 30gb on Onyx. Running beacon with `historical-slasher-node: yes` and slasher with `enable-historical-detection: yes`.

### Has this worked before in a previous version?

Yes, docker tag `HEAD-c26151` worked fine.

## 🔥 Error

```
$ sudo du -sh *
30G     beacon
```

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
Ubuntu 20.04
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
docker tag `HEAD-62ae22`
  </code>
</pre>

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Flag changes

**What does this PR do? Why is it needed?**
This is a PR to modify the historical slasher flag to take up less DB space at the cost of higher CPU usage for historical detection by making it save a archive point every 128 slots vs. the previous 128. This will require a resync for anyone using this flag.
Also fixes the CMD config as some configurations weren't being properly assigned. 

Fixes #6702

### CODE REVIEW

File: beacon-chain/node/node.go [hunk 1]
```diff
@@ -96,13 +96,20 @@ func NewBeaconNode(cliCtx *cli.Context) (*BeaconNode, error) {
...
-		c.SlotsPerArchivedPoint = params.BeaconConfig().SlotsPerEpoch
+		// Save a state every 4 epochs.
+		c.SlotsPerArchivedPoint = params.BeaconConfig().SlotsPerEpoch * 4
 		params.OverrideBeaconConfig(c)
 		cmdConfig := cmd.Get()
+		// Allow up to 4096 attestations at a time to be requested from the beacon nde.
 		cmdConfig.MaxRPCPageSize = int(params.BeaconConfig().SlotsPerEpoch * params.BeaconConfig().MaxAttestations)
-		cmd.Init(cmdConfig)
+		cmd.OverrideConfig(cmdConfig)
+		log.Warnf(
```
Comment (terencechain) [inline beacon-chain/node/node.go:108]: Why is this a `Warn` rather than `Info`?

File: shared/cmd/config.go [hunk 2]
```diff
@@ -20,67 +20,62 @@ var sharedConfig *Flags
...
-	Init(c)
+	OverrideConfig(c)
 	return resetFunc
 }
 
 // ConfigureBeaconChain sets the global config based
 // on what flags are enabled for the beacon-chain client.
 func ConfigureBeaconChain(ctx *cli.Context) {
 	cfg := newConfig(ctx)
-	delay := params.BeaconConfig().GenesisDelay
```
Comment (nisdas) [inline shared/cmd/config.go:47]: please revert all the genesis delay changes, this breaks how we run the node. The default genesis delay is now zero...

Comment (nisdas) [reply shared/cmd/config.go:47]: ok so it seems like it is set here:
https://github.com/prysmaticlabs/prysm/blob/0178c2b2922a323b59b94c9afa3ba1760fd0842d/shared/cmd/config.go#L24 , although this seems a bit brittle, if `sharedConfig` is already initialized we would ignore it. Can you double check it doesn't break genesis for our testnets ?

Comment (0xKiwi) [reply shared/cmd/config.go:47]: Please check newConfig(), the defaults are now set through there. I can confirm with a test.

Comment (0xKiwi) [reply shared/cmd/config.go:47]: Added a test to confirm the default value @nisdas please check

Comment (0xKiwi) [reply shared/cmd/config.go:47]: I also synced a node to head with these changes :+1:

**Pull request merged**